src/databricks/sql/client.py

```diff
@@ -433,6 +439,43 @@ def _determine_parameter_approach(
         else:
             return ParameterApproach.NATIVE
 
+    def _only_dbsql_parameters_in_list(self, params: ListOfParameters) -> bool:
+        """Return True if all members of the list are DbsqlParameter instances"""
+        return all([isinstance(i, DbsqlParameter) for i in params])
+
+    def _all_dbsql_parameters_are_named(self, params: List[DbsqlParameter]) -> bool:
+        """Return True if all members of the list have a non-null .name attribute"""
+        return all([i.name is not None for i in params])
+
+    def _list_of_params_can_use_named_structure(self, params: ListOfParameters) -> bool:
```

Comment:
when your private/protected methods are named properly, and are obvious (i.e. one liners like here), doc strings are superfluous.  No need to delete what you have, but when you've gone to trouble of making the code clean, you don't need as much doc.